### PR TITLE
azure - add event subscription resource with delete action

### PR DESCRIPTION
**Pull request by erwelch**
Adds the event subscription resource with the delete action.

### CODE REVIEW

File: tools/c7n_azure/tests/test_event_subscriptions.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    def setUp(self):
+        super(AzureEventSubscriptionsTest, self).setUp()
+
+    def _create_event_subscription(self):
+        account = self.setup_account()
+        queue_name = 'cctesteventsub'
+        StorageUtilities.create_queue_from_storage_account(account, queue_name)
+        event_sub_destination = StorageQueueEventSubscriptionDestination(
+            resource_id=account.id, queue_name=queue_name)
+        event_sub_name = 'custodiantestsubscription'
```
Comment (logachev) [inline tools/c7n_azure/tests/test_event_subscriptions.py]: nit: you can move it to the class constant string (and no need to return it from function)
(same with queue name)

File: tools/c7n_azure/tests/test_event_subscriptions.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        AzureEventSubscription.create(event_sub_destination, event_sub_name)
+        return event_sub_name
+
+    @arm_template('storage.json')
+    def test_azure_event_subscription_policy_run(self):
+        event_sub_name = self._create_event_subscription()
```
Comment (logachev) [inline tools/c7n_azure/tests/test_event_subscriptions.py]: nit: you can move it to setUp function

File: tools/c7n_azure/c7n_azure/resources/event_subscription.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            'properties.topic'
+        )
+
+
+@EventSubscription.action_registry.register('delete')
+class Delete(BaseAction):
+    schema = type_schema('delete')
+
+    def process(self, resources, event=None):
+        return ThreadHelper.execute_in_parallel(resources=resources,
```
Comment (logachev) [inline tools/c7n_azure/c7n_azure/resources/event_subscription.py]: probably out of scope for this PR, but we can move this execute_in_parallel to some AzureBaseAction, so all Azure actions have it

File: tools/c7n_azure/tests/test_event_subscriptions.py (hunk 2 above)
Comment (erwelch) [reply tools/c7n_azure/tests/test_event_subscriptions.py]: Oh yeah definitely - I had a separate test that tested validation and didn't require an event subscription existing, but removed it. I'll put it in setUp!

File: tools/c7n_azure/c7n_azure/resources/event_subscription.py (hunk 3 above)
Comment (erwelch) [reply tools/c7n_azure/c7n_azure/resources/event_subscription.py]: Good idea! I'll follow-up with a PR that does that and refactors all the other parallel calls we make (or should be making).

**Pull request merged**